pandas/tests/internals/test_internals.py

```diff
@@ -1216,7 +1238,17 @@ def test_binop_other(self, op, value, dtype):
         }
 
         if (op, dtype) in invalid:
-            with pytest.raises(TypeError):
+            msg = (
+                r"cannot perform __pow\_\_ with this index type: DatetimeArray|"
+                r"cannot perform __mod\_\_ with this index type: DatetimeArray|"
+                r"cannot perform __truediv\_\_ with this index type: DatetimeArray|"
+                r"cannot perform __mul\_\_ with this index type: DatetimeArray|"
+                r"cannot perform __pow\_\_ with this index type: TimedeltaArray|"
+                "ufunc 'multiply' cannot use operands with types dtype"
+                r"\('<m8\[ns\]'\) and dtype\('<m8\[ns\]'\)|"
+                "cannot add DatetimeArray and Timestamp"
```

Comment:
Personally, I am -1 on adding message asserts like this, if there are that many as this (and some are not even good error messages). Some of them are also out of our control (numpy error messages)